src/pegen/parser.py

```diff
@@ -15,6 +16,14 @@
 F = TypeVar("F", bound=Callable[..., Any])
 
 
+@final
+class NotParsedMarker(Enum):
+    marker = 0
+
+
+not_parsed = NotParsedMarker.marker
```

Comment:
You can just use `object()` and it'll give the same effect
Also `@final` doesn't do a lot, It's better to just use `Literal[not_parsed]` for annotations